File: paleale/src/service/profile/protocols.ts
export interface GetProfileResponse {
  firstName: string;
  lastName: string;
  profilePicture: string;
  email: string;
  isConnectedWithUntappd: boolean;
}

export interface AddProfileParams {
  firstName: string;
  lastName: string;
  profilePicture: string;
  email: string;
}

export interface AuthorizeUntappdParams {
  code: string;
}

File: kloster/modules/profile/protocols.go
package profile

type getProfileResponse struct {
	FirstName              string `json:"firstName"`
	LastName               string `json:"lastName"`
	ProfilePicture         string `json:"profilePicture"`
	Email                  string `json:"email"`
	IsConnectedWithUntappd bool   `json:"isConnectedWithUntappd"`
}

type addProfileParams struct {
	FirstName      string `json:"firstName"`
	LastName       string `json:"lastName"`
	ProfilePicture string `json:"profilePicture"`
	Email          string `json:"email"`
}

type authorizeUntappdParams struct {
	Code string `json:"code"`
}
